pkg/streamingpromql/operators/binary_operation.go

```diff
@@ -362,12 +365,39 @@ func (b *BinaryOperation) labelsFunc() func(labels.Labels) labels.Labels {
 
 	return func(l labels.Labels) labels.Labels {
 		lb.Reset(l)
-		lb.Del(b.VectorMatching.MatchingLabels...)
 		lb.Del(labels.MetricName)
+		lb.Del(b.VectorMatching.MatchingLabels...)
 		return lb.Labels()
 	}
 }
 
+// groupKeyFunc returns a function that computes the grouping key of the output group this series belongs to.
+func (b *BinaryOperation) groupKeyFunc() func(labels.Labels) []byte {
```

Comment:
would be good to see some tests of this to make sure the fast paths work as expected etc

Reply:
This should be covered by the existing `.test` test cases, which is why I didn't add any further tests here.